gbulb/glib_events.py

```diff
@@ -725,12 +712,10 @@ def run_until_complete(self, future, **kw):
         def stop(f):
             self.stop()
 
-        future = tasks.async(future, loop=self)
+        future = tasks.ensure_future(future, loop=self)
         future.add_done_callback(stop)
-        try:
-            self.run_forever(**kw)
-        finally:
-            future.remove_done_callback(stop)
```

Comment:
Is there a reason you've dropped the try/finally here?

Reply:
@nathan-hoad Great catch, I'm not sure why I removed that, but it looks like it was a mistake. I have pushed a new commit that restores the Try/Finally. :+1: